File: count-binary-substrings/count_binary_substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var countBinarySubstrings = function(s) {
    var starti = 0, lasti = 0, sum = 0;
    for (var i = 1; i < s.length; i ++) {
        if (s.charAt(i) != s.charAt(i-1)) {
            sum += 1;
            lasti = i - starti;
            starti = i;
        } else if (i - starti < lasti) {
            sum += 1;
        }
    }
    return sum;
};

File: count-binary-substrings/count_binary_substrings.go
func countBinarySubstrings(s string) int {
    var starti, lasti, sum int
    for i := 1; i < len(s); i ++ {
        if s[i] != s[i-1] {
            sum += 1
            lasti = i - starti
            starti = i
        } else if i - starti < lasti {
            sum += 1
        }
    }
    return sum
}